packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -6088,6 +6088,17 @@ export default class Meeting extends StatelessWebexPlugin {
         body.resourceToken = resourceToken;
       }
 
+      // @ts-ignore
+      this.webex.internal.newMetrics.submitClientEvent({
+        name: 'client.share.floor-grant.request',
```

Comment:
in stopWhiteboardShare() we are sending "client.share.stopped", but here we're sending "client.share.floor-grant.request" - the names of the events  don't seem to be very consistent, so just wanted to double check that this is what Call Analyzer wants, is there a spec somewhere that confirms this?

Reply:
https://confluence-eng-gpk2.cisco.com/conf/pages/viewpage.action?spaceKey=MS&title=Share+Analyzer+Details

This is the wiki I'm going off of. Are we certain that `client.share.stopped` should be sent the way it is in `stopWhiteboardShare()`? To me it seems like that would be the response of a successful request, in that case we probably should send `client.share.floor-grant.request` before the request. Again, will remove so we can discuss for a different time.

Reply:
> This is the wiki I'm going off of.

cool, that wiki looks very useful. A few things I've noticed from it:
1. we need to generate the `shareInstanceId` and send it with all the share related events, without this, all of them will be ignored by CA. I don't see `shareInstanceId` being generated or sent in the current code or in this PR. Is the plan to do it in another PR or should it be added here?
2.  wiki has a client event `client.share.floor-accept.notification` - again don't see it in the code or in this PR, should it be added here or in another PR? actually I don't understand what's the difference between that event and `client.share.floor-granted.local` which we already send, maybe we don't need to send `client.share.floor-accept.notification`? Can you confirm with Meera?
3. wiki mentions also `client.media.tx.start` and `client.media.tx.stop` events. We currently don't send them for share - see the calls to emitStartStopEvents() in statsAnalyzer/index.ts - I think we will need to get that fixed, maybe should be a separate PR so that this one doesn't grow too big, please raise a Jira for this and add to the epic

> Again, will remove so we can discuss for a different time.
ok, but then do we have a jira for handling metrics for whiteboard? I guess it should still be in the same "share" epic

Reply:
1. This wasn't discussed with Meera but you're totally right, I will ask about this.
2. `client.share.floor-accept.notification` is mostly used for whiteboard, I was told. So we can come back to that for the future epic
3. `client.media.tx.start` and `client.media.tx.stop` are being sent in `setupStatsAnalyzerEventHandlers()`, should they be sent from somewhere else instead?

Reply:
I will create follow up PR with shareinstanceid work

Reply:
> client.media.tx.start and client.media.tx.stop are being sent in setupStatsAnalyzerEventHandlers(), should they be sent from somewhere else instead?

yes, but if you follow the code, you'll see that they're sent when stats analyzer emits LOCAL_MEDIA_STARTED/LOCAL_MEDIA_STOPPED events and these are only sent from emitStartStopEvents() and that function is only called for main audio and video, but not for share

Reply:
Updated https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-471989 which is in the follow up epic